disttask/framework/dispatcher/dispatcher.go

```diff
@@ -65,6 +66,8 @@ type TaskHandle interface {
 	GetAllSchedulerIDs(ctx context.Context, gTaskID int64) ([]string, error)
 	// GetPreviousSubtaskMetas gets previous subtask metas.
 	GetPreviousSubtaskMetas(gTaskID int64, step int64) ([][]byte, error)
+	// WithNewSession executes the function with a new session.
+	WithNewSession(fn func(se sessionctx.Context) error) error
```

Comment:
```suggestion
	ExecInNewSession(fn func(se sessionctx.Context) error) error
```

Reply:
it only provides the session, bussiness decide whether to execute or not

Reply:
WithNewSession like an option function used by NewXXX whose argument is a session, and let XXX use the session in argument.

Reply:
[ffca1de](https://github.com/pingcap/tidb/pull/44198/commits/ffca1de23a5790ebaf89d2dd3f053b5899fa0c45)